Project-Lily/Project-Lily/NetworkManager.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  NetworkManager.swift
+//  Project-Lily
+//
+//  Created by Kendall Poindexter on 11/16/18.
+//  Copyright © 2018 Kendall Poindexter. All rights reserved.
+//
+
+import Foundation
+
+class NetworkManager {
+    
+    private let urlString = "https://private-52aac-breeds1.apiary-mock.com/breeds"
+    
+    func createDogURLSession(completion: @escaping ([Dog]) -> Void) {
+        let session = URLSession.shared
+        guard let url = createDogURL(urlString: urlString) else { return }
+        let dataTask = session.dataTask(with: url) { (data, response, error) in
+            if let error = error {
+                print("Failure! \(error)")
+            } else if let httpResponse = response as? HTTPURLResponse,
+                httpResponse.statusCode == 200 {
+                if let data = data {
+                    let dogs: [Dog] = self.parse(data: data) ?? []
+                   
+                    
+                    DispatchQueue.main.async {
```

Comment:
🍰 It might be a good idea to only dispatch to the main thread wherever this function gets called. The reasoning is because there might be a place you want to call this function and—in the completion handler—you *don't* necessarily want that code on the main thread. 

By only dispatching to the main thread from within the completion blocks themselves (wherever you call this function), you make the behavior more customizable to various needs.